### ISSUE TITLE
Paths starting with `./` in DefaultItemExcludes are not properly handled

### ISSUE THREAD

**Issue by lanfeust69** [microsoft/msbuild#5281]
This is probably a flaw in the implementation of [FileMatcher](/microsoft/msbuild/blob/vs16.5/src/Shared/FileMatcher.cs).

### Steps to reproduce

See [here](https://github.com/lanfeust69/TestDefaultItemExcludes)

```
msbuild /bl /p:WithDot=True
```
### Expected behavior

`None` items should only contain the `.git*` files and `ReadMe.md`

### Actual behavior

`None` items also include the `bin` and `Data` contents.

### Environment data
`msbuild /version` output:
`16.5.0.12403`

OS info:
both Windows 10 and Ubuntu 19.10

**Comment by Forgind** [comment]
Thanks for bringing this to our attention. Having looked at your project file, I suspect the problem is that BaseOutputPath, despite what our documentation says, isn't properly supported. You might be interested in the PR to fix that [here](https://github.com/microsoft/msbuild/pull/5238) and the issue it references [here](https://github.com/microsoft/msbuild/issues/1664). I'm going to close this in favor of that issue, but after we've resolved it, if you're still hitting this problem, let me know, and I can reopen this and dig deeper.

**Comment by Forgind** [comment]
BaseOutputPath isn't actually necessary. I'll look more soon.

**Comment by lanfeust69** [comment]
> BaseOutputPath isn't actually necessary. I'll look more soon.

Indeed, the project just mentioned two cases where the issue could happen.

**Comment by rainersigwald** [comment]
@lanfeust69 It's reasonable to start supporting this, but the FileMatcher often expects exact string prefix matches, so dropping `./` is helpful today.
